src/aliceVision/matchingImageCollection/GeometricFilterMatrix_HGrowing.hpp

```diff
@@ -0,0 +1,345 @@
+ * @param[in] outFilename The filename of the svg file to generate.
+ * @param[in] viewI The first view.
+ * @param[in] viewJ The second view.
+ * @param[in] featuresI The features of the first view.
+ * @param[in] featuresJ The features of the second view.
+ * @param[in] homographiesAndMatches Contains each found homography and the relevant supporting matches.
+ * @param[in] putativeMatches The putative matches.
+ */
+void drawHomographyMatches(const std::string& outFilename,
+                      const sfm::View & viewI,
+                      const sfm::View & viewJ,
+                      const std::vector<feature::SIOPointFeature>& siofeatures_I,
+                      const std::vector<feature::SIOPointFeature>& siofeatures_J,
+                      const std::vector<std::pair<Mat3, matching::IndMatches>>& homographiesAndMatches,
+                      const matching::IndMatches& putativeMatches);
+
+//-- Multiple homography matrices estimation template functor, based on homography growing, used for filter pair of putative correspondences
+struct GeometricFilterMatrix_HGrowing : public GeometricFilterMatrix
+{
+    explicit GeometricFilterMatrix_HGrowing(
+          double dPrecision = std::numeric_limits<double>::infinity(),
+          size_t iteration = 1024)
+              : GeometricFilterMatrix(dPrecision, std::numeric_limits<double>::infinity(), iteration)
+  { }
+  
+  /**
+   * @brief Given two sets of image points, it estimates the homography matrix
+   * relating them using a robust method (like A Contrario Ransac).
+   * @details It return matches grouped by estimated homographies.
+   * Example:
+   * - Putative matches id:         [0 1 2 3 4 5 6 7]
+   * - Estimated homographies:      [0(h0) 1(h2) 2(h1) 3(nan) 4(h2) 5(h0) 6(h1) 7(nan)]
+   * - Sorted matches               [0(h0) 5(h0) 2(h1) 6(h1) 1(h2) 4(h2)]
+   * - out_geometricInliersPerType  [0 5 2 6 1 4]
+   */
+  template<typename Regions_or_Features_ProviderT>
+  EstimationStatus geometricEstimation(const sfm::SfMData * sfmData,
+                                       const Regions_or_Features_ProviderT &regionsPerView,
+                                       const Pair &pairIndex,
+                                       const matching::MatchesPerDescType &putativeMatchesPerType,
+                                       matching::MatchesPerDescType &out_geometricInliersPerType)
+  {
+    using namespace aliceVision::feature;
+    using namespace aliceVision::matching;
+    
+    out_geometricInliersPerType.clear();
+    
+    // ------------------------
+    // To draw & save matches groups into .svg images:
+    //    enter an existing folder in the following variable ('outputSvgDir'). 
+    // File format: <nbHMatches>hmatches_<viewId_I>_<viewId_J>_<descType>.svg
+    // * Little white dots = putative matches
+    // * Colored dots = geometrically verified matches (1 color per estimated plane)
+    std::string outputSvgDir;
```

Comment:
Actually, since this is empty, all the part saving the result will never be called.
Shall we add it as a param of the method with default value `""` (otherwise I guess we break the template call from the other classes) or shall we just leave it as a debug stuff